base/task.jl

```diff
@@ -972,7 +978,13 @@ function enq_work(t::Task)
     return t
 end
 
-schedule(t::Task) = enq_work(t)
+function schedule(t::Task)
+    # user_time -task-(re)scheduled-> wait_time
+    if t.is_timing_enabled && t.first_enqueued_at == 0
+        t.first_enqueued_at = time_ns()
+    end
```

Comment:
Thanks for having these edge labels, they're very helpful. ❤️ 

If in a future PR we add a distinction between wait-reasons, we would probably put another update here to take us from e.g. waiting on a lock to waiting in the run queue.

🤔 ... which actually makes me think: I think this isn't actually taking the Task `t` from user -> wait_time. The task `t` wasn't running before this function was called.

This function is moving the task from one kind of not-running to a new kind of not-running. In this PR, it's moving it from state CREATED to state WAITING, which we didn't originally include in our state diagram, but i guess we should have.

```suggestion
    # created -task-scheduled-> wait_time
    if t.is_timing_enabled && t.first_enqueued_at == 0
        t.first_enqueued_at = time_ns()
    end
```
And i think in the future, we'd add another possible edge for this:
```
    # waiting on lock -task-scheduled-> waiting in run queue
```
when we add those finer-grained information.

Reply:
yeah, this should be two arrows, as it can be either being put in the run queue for the first time or because it's being reinserted after e.g. being in `wait(condition)`.

so maybe
```
# created -task-scheduled-> wait_time, or
# blocked_on_conditon -task-scheduled-> wait_time
```
?

Reply:
Yeah, exactly.

But i would not add the second arrow yet, since we aren't actually recording anything about that edge, right? Your if-condition is only checking for the first arrow.

And in particular, the states on either side of the arrow, "blocked_on_condition" and "in the run queue" are _both_ "wait_time", so i think that we would only want to add an arrow here once we have broken this down into more detailed states.